independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateInstance.java

```diff
@@ -178,6 +179,48 @@ default TemplateInstance onRendered(Runnable action) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Sets the {@code locale} attribute that can be used to localize parts of the template, i.e. to specify the locale for all
+     * message bundle expressions in the template.
+     *
+     * @param locale a language tag
+     * @return self
+     */
+    default TemplateInstance setLocale(String locale) {
+        return setAttribute(LOCALE, Locale.forLanguageTag(locale));
+    }
+
+    /**
+     * Sets the {@code locale} attribute that can be used to localize parts of the template, i.e. to specify the locale for all
+     * message bundle expressions in the template.
+     *
+     * @param locale a {@link Locale} instance
+     * @return self
+     */
+    default TemplateInstance setLocale(Locale locale) {
+        return setAttribute(LOCALE, locale);
+    }
+
+    /**
+     * Sets the variant attribute that can be used to select a specific variant of the template.
+     *
+     * @param variant the variant
+     * @return self
+     */
+    default TemplateInstance setVariant(Variant variant) {
+        return setAttribute(SELECTED_VARIANT, variant);
+    }
+
+    /**
+     * Sets the variants attribute that can be used to select a specific variant of the template.
+     *
+     * @param variants the variants
+     * @return self
+     */
+    default TemplateInstance setVariants(Iterable<Variant> variants) {
```

Comment:
Hm, this method should never be called by users. The variants are set by the engine. It might make sense to obtain the set of variants but it would be better to skip this attribute altogether :shrug: .

Reply:
Gotcha, I'll remove it